OsmAnd/src/net/osmand/plus/AppInitializer.java

```diff
@@ -228,8 +228,20 @@ public void onFinish(AppInitializer init) {
 				app.getSettings().migratePreferences();
 				startPrefs.edit().putInt(VERSION_INSTALLED_NUMBER, VERSION_3_6).commit();
 			}
-			if (prevAppVersion < VERSION_3_7) {
+			if (prevAppVersion < VERSION_3_7 || Version.getAppVersion(app).equals("3.7.1")) {
```

Comment:
Looks bad
Why it is not simply prevAppVersion < VERSION_3_7_1

Reply:
// Make 4 digits version  from now
	public static final int VERSION_3_7_1 = 3710;

Reply:
Move all migration code to Another class AppVersionUpgradeOnInit